library/src/main/java/com/cube/storm/ui/fragment/StormBottomTabsFragment.java

```diff
@@ -82,32 +116,99 @@ public void onLoadingComplete(
 		}
 	}
 
+	@Override public void onCreateOptionsMenu(Menu menu, MenuInflater inflater)
+	{
+		super.onCreateOptionsMenu(menu, inflater);
+		// It will only create a menu item if the number of pages is greater than 5
+		if (viewPager.getAdapter() != null && viewPager.getAdapter().getCount() > MAX_BOTTOM_TABS)
+		{
+			for (int iterator = MAX_BOTTOM_TABS; iterator < pageAdapter.getPages().size(); iterator++)
+			{
+				MenuItem item = menu.add(viewPager.getAdapter().getPageTitle(iterator));
+				final int finalIterator = iterator;
+				item.setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener()
+				{
+					@Override
+					public boolean onMenuItemClick(MenuItem item)
+					{
+						if (getContext() == null)
+						{
+							return false;
+						}
+						PageDescriptor pageDescriptor = pageAdapter.getPages().get(finalIterator).getPageDescriptor();
+						Intent stormIntent = UiSettings.getInstance().getIntentFactory().getIntentForPageDescriptor(getContext(), pageDescriptor);
+						startActivity(stormIntent);
+						return true;
+					}
+				});
+			}
+		}
+	}
+
 	@Override public int getLayoutResource()
 	{
 		return R.layout.tabbed_page_bottom_fragment_view;
 	}
 
 	@Override
-	public boolean onTabSelected(
-		int position,
-		boolean wasSelected
-	)
+	public boolean onTabSelected(int position, boolean wasSelected)
 	{
-		AHBottomNavigationItem item = bottomNavigation.getItem(position);
 		viewPager.setCurrentItem(position);
+		selectedTab = position;
+		setTabItemContentDescriptions();
 
-		final Activity activity = getActivity();
-		if (activity != null)
-		{
-			activity.setTitle(item.getTitle(activity));
+		if (getActivity() != null && ((AppCompatActivity)getActivity()).getSupportActionBar() != null)
+		{    //Hide the back arrow in the main activity
+			ActionBar actionBar = ((AppCompatActivity)getActivity()).getSupportActionBar();
+			if (actionBar != null)
+			{
+				actionBar.setDisplayHomeAsUpEnabled(false);
+				actionBar.setTitle(bottomNavigation.getItem(position).getTitle(getContext()));
+			}
 		}
-
 		return true;
 	}
 
+	private void setTabItemContentDescriptions()
+	{
+		// Wait for view to be layed out
```

Comment:
Please expand this comment to explain _why_ you do this

Reply:
Yes, I will 😃